realm/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -38,6 +38,9 @@
 
     @Override
     protected void setUp() throws Exception {
+        for (Integer key : Realm.openRealms.keySet()) {
```

Comment:
Having this seem to indicate that some other unit tests are not behaving like they should? Perhaps add a comment about that.